engine/src/audio/SceneManager.cpp

```diff
@@ -458,6 +458,7 @@ void SceneManager::activationPhaseImpl() {
             it != data->activeScenes.end();
             ++it) {
         SimpleScene *scene = dynamic_cast<SimpleScene *>(it->second.get());
+        assert(scene != nullptr);
```

Comment:
Not a macro. Macros bad. No more macros :-)

An inline function would probably be fine though. I like the one you drafted above. That one looks like it would work. Or perhaps just a function to which we pass the original object and the desired new type, directly using template argument(s).

Reply:
@stephengtuggy templates could work too; a number of things I came across while writing that function combined the lambda with a template method. Just looking for graceful handling.

BTW, on a related note - we probably should come up with some kind of handler that can suggest how a user files an Issue when they hit bugs...I'll write up a story for that.

Reply:
There, see what you think of my latest implementation.